check_sites.py

```diff
@@ -85,8 +88,16 @@ def main():
         # If the schema is invalid, we will not run any other checks
         print(inst)
         return
+    
+    # Check for exclusivity among all sets in the updated version
+    try:
+        fps_checker.check_exclusivity(fps_checker.load_sets())
+    except Exception as inst:
+            error_texts.append(inst)
```

Comment:
Why are we moving this check from its original location in the list of checks?

Reply:
As I was editing this I realized that keeping it within the `with_diff` portion meant that the check was only looking at the exclusivity within the new sets portion, essentially defeating the purpose of the check.

Reply:
Ah, great point!

Reply:
Can you write a test case that exercises this? I.e. one that ought to fail the checks, but would pass if you hadn't moved this one.

Reply:
I suppose I can do that, but since this is a change in `check_sites.py` I think I would have to create a new test file that executes the script with different input files (or spoofed versions of input files)? That might make this PR a bit longer. I understand wanting to have the change explicitly tested, and I did test it locally with made up files, just not sure if there's a better way to create an automated test for this type of change?

Reply:
Does `check_sets.py` have to be the thing that knows how to find/parse the input files *and* knows which checks to run? I think we could rearrange things so that check_sets.py is responsible for reading/parsing the input file(s) and CLI options, but then hands the parsed input off to something else to actually do the validation and create the appropriate errors. (And that way we could have unit tests for all of the validation steps, and the tests wouldn't have to worry about input files - they'd just supply the desired hardcoded inputs to the thing that does the validation.)

I agree with you though that that seems like a bigger refactoring and probably doesn't belong in this CL. (Strictly speaking, each CL/PR should make only one change, so fixing the disjointness check belongs in a separate PR. But it's already written, and small, so I won't make you move it now.)